collectors/metrics/pkg/metricfamily/label.go

```diff
@@ -53,21 +53,44 @@ func appendLabels(
 	existing []*clientmodel.LabelPair,
 	overrides map[string]*clientmodel.LabelPair) []*clientmodel.LabelPair {
 	var found []string
+
+	// override matching existing labels
 	for i, pair := range existing {
 		name := pair.GetName()
+		// remove any name == "" or value == nil
+		if name == "" || pair != nil {
+			existing = append(existing[:i], existing[i+1:]...)
+		}
 		if value, ok := overrides[name]; ok {
 			existing[i] = value
 			found = append(found, name)
 		}
 	}
+
+	// append any overrides that didn't already exist
 	for k, v := range overrides {
-		if !contains(found, k) {
-			existing = append(existing, v)
+		// only append real names and values
+		// don't append anything overwritten above
+		if k != "" && v != nil && !contains(found, k) {
+			existing = insertLexicographicallyByName(existing, v)
 		}
 	}
 	return existing
 }
 
+func insertLexicographicallyByName(
```

Comment:
we also need this function in https://github.com/stolostron/multicluster-observability-operator/blob/072204b63ed0fab76611e21a8d5b63a274afbbc3/collectors/metrics/pkg/metricfamily/hypershift_transformer.go#L81-L121